[CTS] API response is not unmarshalled properly

### Terraform Version

```
Terraform v1.1.7
on linux_amd64
+ provider registry.terraform.io/toolbox/flexibleengine v1.28.0
```

### Affected Resource(s)

- flexibleengine_cts_tracker_v1

### Terraform Configuration Files

```hcl
resource "flexibleengine_cts_tracker_v1" "tracker" {
  bucket_name = "xxxx"
  status      = "enabled"
  region      = "eu-west-0"
}
```

The tracker and bucket was created some time ago. Then plans started to fail with:

```
Stack trace from the terraform-provider-flexibleengine_v1.25.1 plugin:
panic: runtime error: index out of range [0] with length 0

goroutine 90 [running]:
github.com/terraform-providers/terraform-provider-flexibleengine/flexibleengine.resourceCTSTrackerRead(0xc0004f9480, 0x1454740, 0xc00000c1e0, 0x1eefff0, 0xc000100800)
github.com/terraform-providers/terraform-provider-flexibleengine/flexibleengine/resource_flexibleengine_cts_tracker_v1.go:108 +0x88f
```

I saw that this was fixed in 1.28.0 via c4e39c8d6ea1183a6eb85fd1174f582055ee65bb, but now with 1.28.0 I get this:

```
flexibleengine_cts_tracker_v1.tracker: Refreshing state... [id=system]

Error: can not find CTS tracker system

with flexibleengine_cts_tracker_v1.tracker,
on main.tf line 1, in resource "flexibleengine_cts_tracker_v1" "tracker":
resource "flexibleengine_cts_tracker_v1" "tracker" {
```

### Debug Output

Looks like the API response is not what is expected:

```
2022-04-12T08:50:28.496+0200 [INFO]  provider.terraform-provider-flexibleengine_1.28.0: 2022/04/12 08:50:28 [DEBUG] Unable to parse JSON: json: cannot unmarshal array into Go value of type map[string]interface {}: timestamp=2022-04-12T08:50:28.496+0200
2022-04-12T08:50:28.496+0200 [INFO]  provider.terraform-provider-flexibleengine_1.28.0: 2022/04/12 08:50:28 [DEBUG] API Response Body: [{"bucket_name":"xxxxx","tracker_name":"system","detail":"noBucket","file_prefix_name":"","status":"error"}]: timestamp=2022-04-12T08:50:28.496+0200
2022-04-12T08:50:28.497+0200 [ERROR] vertex "flexibleengine_cts_tracker_v1.tracker" error: can not find CTS tracker system
```

Related #727

@ShiChangkuo here's what I can find:
 
```
flexibleengine_s3_bucket_policy.bucket_policies: Refreshing state... [id=xxxxx]
flexibleengine_cts_tracker_v1.tracker: Refreshing state... [id=system]
2022-04-14T09:12:30.251+0200 [INFO]  provider.terraform-provider-flexibleengine_1.28.0: 2022/04/14 09:12:30 [DEBUG] initialize Swift S3 session: timestamp=2022-04-14T09:12:30.251+0200
2022-04-14T09:12:30.251+0200 [INFO]  provider.terraform-provider-flexibleengine_1.28.0: 2022/04/14 09:12:30 [INFO] Building Swift S3 auth structure: timestamp=2022-04-14T09:12:30.251+0200
2022-04-14T09:12:30.251+0200 [INFO]  provider.terraform-provider-flexibleengine_1.28.0: 2022/04/14 09:12:30 [INFO] Setting AWS metadata API timeout to 100ms: timestamp=2022-04-14T09:12:30.251+0200
2022-04-14T09:12:30.251+0200 [INFO]  provider.terraform-provider-flexibleengine_1.28.0: 2022/04/14 09:12:30 [DEBUG] API Request URL: GET https://cts.eu-west-0.prod-cloud-ocb.orange-business.com/v1.0/9f2e36b98a1149aab266626ffbbfXXXX/tracker: timestamp=2022-04-14T09:12:30.251+0200
2022-04-14T09:12:30.251+0200 [INFO]  provider.terraform-provider-flexibleengine_1.28.0: 2022/04/14 09:12:30 [DEBUG] API Request Headers:
Accept: application/json
User-Agent: terraform-provider-flexibleengine golangsdk/2.0.0
X-Auth-Token: ***: timestamp=2022-04-14T09:12:30.251+0200
2022-04-14T09:12:30.726+0200 [INFO]  provider.terraform-provider-flexibleengine_1.28.0: 2022/04/14 09:12:30 [DEBUG] API Response Code: 200: timestamp=2022-04-14T09:12:30.726+0200
2022-04-14T09:12:30.726+0200 [INFO]  provider.terraform-provider-flexibleengine_1.28.0: 2022/04/14 09:12:30 [DEBUG] API Response Headers:
Connection: keep-alive
Content-Length: 115
Content-Type: application/json
Date: Thu, 14 Apr 2022 07:12:30 GMT
Set-Cookie: Path=/;Secure;HttpOnly
Strict-Transport-Security: max-age=16070400; includeSubDomains
X-Content-Type-Options: nosniff
X-Download-Options: noopen
X-Frame-Options: SAMEORIGIN
X-Request-Id: 8628abb8bcb92ddaaf4e729c28197806
X-Xss-Protection: 1; mode=block;: timestamp=2022-04-14T09:12:30.726+0200
2022-04-14T09:12:30.726+0200 [INFO]  provider.terraform-provider-flexibleengine_1.28.0: 2022/04/14 09:12:30 [DEBUG] Unable to parse JSON: json: cannot unmarshal array into Go value of type map[string]interface {}: timestamp=2022-04-14T09:12:30.726+0200
2022-04-14T09:12:30.726+0200 [INFO]  provider.terraform-provider-flexibleengine_1.28.0: 2022/04/14 09:12:30 [DEBUG] API Response Body: [{"bucket_name":"xxxxx","tracker_name":"system","detail":"noBucket","file_prefix_name":"","status":"error"}]: timestamp=2022-04-14T09:12:30.726+0200
2022-04-14T09:12:30.727+0200 [ERROR] vertex "flexibleengine_cts_tracker_v1.tracker" error: can not find CTS tracker system
2022-04-14T09:12:30.727+0200 [ERROR] vertex "flexibleengine_cts_tracker_v1.tracker" error: can not find CTS tracker system
2022-04-14T09:12:30.727+0200 [ERROR] vertex "flexibleengine_cts_tracker_v1.tracker (expand)" error: can not find CTS tracker system
```
 
I believe the call is: `[DEBUG] API Request URL: GET https://cts.eu-west-0.prod-cloud-ocb.orange-business.com/v1.0/9f2e36b98a1149aab266626ffbbfXXXX/tracker:`
 
Where `9f2e36b98a1149aab266626ffbbfXXXX` is my tenant ID.

--- a/flexibleengine/resource_flexibleengine_as_configuration_v1.go
+++ b/flexibleengine/resource_flexibleengine_as_configuration_v1.go
@@ -223,7 +223,7 @@ func getPublicIps(publicIpMeta map[string]interface{}) configurations.PublicIpOp
 	}
 
 	eipOpts := configurations.EipOpts{
-		IpType:    eipMap["ip_type"].(string),
+		Type:      eipMap["ip_type"].(string),
 		Bandwidth: bandWidthOpts,
 	}
 
@@ -247,7 +247,7 @@ func getInstanceConfig(configDataMap map[string]interface{}) (configurations.Ins
 	log.Printf("[DEBUG] get personality: %#v", personalities)
 
 	instanceConfigOpts := configurations.InstanceConfigOpts{
-		ID:          configDataMap["instance_id"].(string),
+		InstanceID:  configDataMap["instance_id"].(string),
 		FlavorRef:   configDataMap["flavor"].(string),
 		ImageRef:    configDataMap["image"].(string),
 		SSHKey:      configDataMap["key_name"].(string),
@@ -263,7 +263,7 @@ func getInstanceConfig(configDataMap map[string]interface{}) (configurations.Ins
 	if len(pubicIpData) == 1 {
 		publicIpMap := pubicIpData[0].(map[string]interface{})
 		publicIps := getPublicIps(publicIpMap)
-		instanceConfigOpts.PubicIp = publicIps
+		instanceConfigOpts.PubicIP = &publicIps
 		log.Printf("[DEBUG] get publicIps: %#v", publicIps)
 	}
 	log.Printf("[DEBUG] get instanceConfig: %#v", instanceConfigOpts)

--- a/flexibleengine/resource_flexibleengine_cts_tracker_v1.go
+++ b/flexibleengine/resource_flexibleengine_cts_tracker_v1.go
@@ -1,10 +1,9 @@
 package flexibleengine
 
 import (
-	"time"
-
 	"fmt"
 	"log"
+	"time"
 
 	"github.com/chnsz/golangsdk/openstack/cts/v1/tracker"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
@@ -21,8 +20,8 @@ func resourceCTSTrackerV1() *schema.Resource {
 		},
 
 		Timeouts: &schema.ResourceTimeout{
-			Create: schema.DefaultTimeout(10 * time.Minute),
-			Delete: schema.DefaultTimeout(10 * time.Minute),
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
 		},
 
 		Schema: map[string]*schema.Schema{
@@ -88,11 +87,9 @@ func resourceCTSTrackerRead(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("Error creating cts Client: %s", err)
 	}
 
+	trackerName := d.Id()
 	listOpts := tracker.ListOpts{
-		TrackerName:    d.Get("tracker_name").(string),
-		BucketName:     d.Get("bucket_name").(string),
-		FilePrefixName: d.Get("file_prefix_name").(string),
-		Status:         d.Get("status").(string),
+		TrackerName: trackerName,
 	}
 	trackers, err := tracker.List(ctsClient, listOpts)
 	if err != nil {
@@ -100,7 +97,7 @@ func resourceCTSTrackerRead(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	if len(trackers) == 0 {
-		return fmt.Errorf("can not find CTS tracker %s", d.Id())
+		return fmt.Errorf("cannot find CTS tracker %s", trackerName)
 	}
 
 	ctsTracker := trackers[0]
@@ -124,7 +121,7 @@ func resourceCTSTrackerUpdate(d *schema.ResourceData, meta interface{}) error {
 	}
 	var updateOpts tracker.UpdateOpts
 
-	//as bucket_name is mandatory while updating tracker
+	// bucket_name is mandatory while updating tracker
 	updateOpts.BucketName = d.Get("bucket_name").(string)
 
 	if d.HasChange("file_prefix_name") {
@@ -138,7 +135,8 @@ func resourceCTSTrackerUpdate(d *schema.ResourceData, meta interface{}) error {
 	if err != nil {
 		return fmt.Errorf("Error updating cts tracker: %s", err)
 	}
-	time.Sleep(20 * time.Second)
+
+	time.Sleep(10 * time.Second)
 	return resourceCTSTrackerRead(d, meta)
 }
 
@@ -154,7 +152,7 @@ func resourceCTSTrackerDelete(d *schema.ResourceData, meta interface{}) error {
 		return err
 	}
 
-	time.Sleep(20 * time.Second)
+	time.Sleep(10 * time.Second)
 	log.Printf("[DEBUG] Successfully deleted cts tracker %s", d.Id())
 
 	return nil

--- a/go.mod
+++ b/go.mod
@@ -4,12 +4,12 @@ go 1.14
 
 require (
 	github.com/aws/aws-sdk-go v1.25.3
-	github.com/chnsz/golangsdk v0.0.0-20220328032852-6b7cdcd8a506
+	github.com/chnsz/golangsdk v0.0.0-20220414095129-997c3994cc66
 	github.com/hashicorp/errwrap v1.1.0
 	github.com/hashicorp/go-cleanhttp v0.5.2
 	github.com/hashicorp/go-multierror v1.1.1
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.10.0
-	github.com/huaweicloud/terraform-provider-huaweicloud v1.35.0
+	github.com/huaweicloud/terraform-provider-huaweicloud v1.35.2-0.20220413091922-4559afe01d1d
 	github.com/jen20/awspolicyequivalence v1.1.0
 	github.com/jtolds/gls v4.20.0+incompatible // indirect
 	github.com/mitchellh/go-homedir v1.1.0

